trackintel/preprocessing/staypoints.py

```diff
@@ -75,6 +80,10 @@ def generate_locations(
 
     # initialize the return GeoDataFrames
     sp = staypoints.copy()
+    non_activities = None
+    if activities_only:
+        non_activities = sp[~sp["activity"]]
```

Comment:
I am unsure if this will raise an error (and how informative the error message is for this specific function) if `sp` does not have an `activity` column. Do we need to write our own `try...except` statement for the failure situation?

Reply:
Yes, I agree a more informative error would be nicer.